frontend/appflowy_flutter/lib/features/mension_person/logic/mention_bloc.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:appflowy/core/config/kv.dart';
+import 'package:appflowy/core/config/kv_keys.dart';
+import 'package:appflowy/features/mension_person/data/cache/person_list_cache.dart';
+import 'package:appflowy/features/mension_person/data/models/models.dart';
+import 'package:appflowy/features/mension_person/data/repositories/mention_repository.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'mention_bloc.freezed.dart';
+
+// bool _showMorePages = false, _showMorePersons = false;
+
+class MentionBloc extends Bloc<MentionEvent, MentionState> {
+  MentionBloc(this.repository, this.workspaceId)
+      : super(MentionState.initial()) {
+    _personListCache = getIt<PersonListCache>();
+    on<Initial>(_onInitial);
+    on<Query>(_onQuery);
+    on<GetPersons>(_onGetPersons);
+    on<UpdatePersonList>(_onUpdatePersonList);
+    on<ShowMorePersons>(_onShowMorePersons);
+    on<ShowMorePages>(_onShowMorePages);
+    on<ToggleSendNotification>(_onToggleSendNotification);
+    on<AddVisibleItem>(_onAddVisibleItem);
+    on<RemoveVisibleItem>(_onRemoveVisibleItem);
+    on<SelectItem>(_onSelectItem);
+    add(MentionEvent.init());
+    add(MentionEvent.getPersons(workspaceId: workspaceId));
+  }
+  final MentionRepository repository;
+  final String workspaceId;
+  late final PersonListCache _personListCache;
+
+  Future<void> _onInitial(
+    Initial event,
+    Emitter<MentionState> emit,
+  ) async {
+    final sendNotification =
```

Comment:
Don't use getIt directly in the bloc. Move it to the repository so that you can mock the data in unit tests.